extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksProcessor.java

```diff
@@ -227,15 +246,26 @@ private void validateRec(IndexView index, String entityType, ClassInfo classInfo
                 .filter(a -> a.name().toString().endsWith("persistence.Id"))
                 .toList();
 
+        List<AnnotationInstance> fieldsAnnotatedWithRestLinkId = classInfo.fields().stream()
+                .flatMap(f -> f.annotations(RestLinkId.class).stream())
+                .toList();
+
+        // @RestLinkId annotation count > 1 is not allowed
+        if (fieldsAnnotatedWithRestLinkId.size() > 1) {
+            throw new IllegalStateException("Cannot generate web links for the class " + entityType +
+                    " because it has multiple fields annotated with `@RestLinkId`, where a maximum of one is allowed");
+        }
+
         // Id field found, break the loop
-        if (!fieldsNamedId.isEmpty() || !fieldsAnnotatedWithId.isEmpty())
+        if (!fieldsNamedId.isEmpty() || !fieldsAnnotatedWithId.isEmpty() || !fieldsAnnotatedWithRestLinkId.isEmpty()) {
             return;
+        }
 
         // Id field not found and hope is gone
         DotName superClassName = classInfo.superName();
         if (superClassName == null) {
             throw new IllegalStateException("Cannot generate web links for the class " + entityType +
-                    " because is either missing an `id` field or a field with an `@Id` annotation");
+                    " because it is either missing an `id` field, a field with an `@Id` annotation or a field with an `@RestLinkId annotation");
```

Comment:
```suggestion
                    " because it is either missing an `id` field, a field with an `@Id` annotation or a field with a `@RestLinkId annotation");
```

Reply:
I'm not a native speaker and not sure if @ counts a vowel or a consonant. I assumed it is as the original author used an.